.github/workflows/ci.yaml

```
@@ -1,244 +1,187 @@
+
+
+  build:
+    name: Build And Test
+    needs: install-dependencies
+    # do not do anything unless changed package is detected
+    if: ${{ needs.install-dependencies.outputs.filter != '{()}' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Source Code
+        uses: actions/checkout@v3
+
+      - name: Setup PNPM with Dependencies
+        uses: ./.github/actions/setup-pnpm-with-dependencies/
+
+      - name: Stylelint Packages
+        run: pnpm --stream --filter "${{needs.install-dependencies.outputs.filter}}" run stylelint
+
+      - name: Lint Packages
+        run: pnpm --stream --filter "${{needs.install-dependencies.outputs.filter}}" run lint
+
+      - name: Build Packages
+        # The `...` syntax tells pnpm to include dependent packages
+        run: pnpm --stream --filter "...${{needs.install-dependencies.outputs.filter}}..." run build
+
+      - name: Run Unit Tests
+        run: pnpm --stream --filter "${{needs.install-dependencies.outputs.filter}}" run test:unit
+
+      - name: Prepare Cypress
+        if: ${{ needs.install-dependencies.outputs.spec != '' }}
+        run: pnpm cypress install
+
+
+      - name: Run Component Tests
+        if: ${{ needs.install-dependencies.outputs.spec != '' }}
+        uses: cypress-io/github-action@v4
+        with:
+          install: false
+          command: pnpm run test:component:ci ${{ needs.install-dependencies.outputs.spec }}
+
+      - name: Save Build Artifacts
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
+        uses: actions/upload-artifact@v3
+        with:
+          name: package-dist-artifacts
+          path: packages/*/*/dist
+
+  publish:
+    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
+    name: Publish
+    needs:
+      - install-dependencies
+      - build
+    runs-on: ubuntu-latest
+    env:
+      GH_TOKEN: ${{ secrets.KONGPONENTS_BOT_PAT }}
+
+    steps:
+      - name: Checkout Source Code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.KONGPONENTS_BOT_PAT }}
+
+      - name: Setup PNPM with Dependencies
+        uses: ./.github/actions/setup-pnpm-with-dependencies/
+
+      - name: Download build artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: package-dist-artifacts
+          path: packages/
+
+      - name: Update package versions
+        run: |
+          git config user.email "konnectx-engineers+kongponents-bot@konghq.com"
+          git config user.name "Kong UI Bot"
+          lerna version --conventional-commits --create-release github --yes
+
+      - name: Publish packages to NPM
+        run: |
+          echo "//registry.npmjs.org/:_authToken=${NODE_AUTH_TOKEN}" > .npmrc
+          pnpm --stream --filter "${{needs.install-dependencies.outputs.filter}}" publish  --no-git-checks --access public
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_PUBLISH_TOKEN }}
```

suggestion: can we flip the order here purely so it's easier to see the `env` vars before the command?
```suggestion
        env:
          NODE_AUTH_TOKEN: ${{ secrets.NPM_PUBLISH_TOKEN }}
        run: |
          echo "//registry.npmjs.org/:_authToken=${NODE_AUTH_TOKEN}" > .npmrc
          pnpm --stream --filter "${{needs.install-dependencies.outputs.filter}}" publish  --no-git-checks --access public
```